### PR TITLE
`dev` feature enabled when compiling without `--release`

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,15 +12,22 @@ rustc-serialize = "0.3"
...
 clippy = { version = "0.0.44", optional = true }
-ethcore-util = { path = "util" }
+
 ethcore = { path = "ethcore" }
+ethcore-util = { path = "util" }
 ethsync = { path = "sync" }
-ethcore-rpc = { path = "rpc", optional = true }
-fdlimit = { path = "util/fdlimit" }
-daemonize = "0.2"
 ethcore-devtools = { path = "devtools" }
```
Comment (arkpar) [inline Cargo.toml]: @NikVolf  Is this a dev dependency as well?

Comment (tomusdrw) [inline Cargo.toml]: I think that it isn't. It should still be used when you run `cargo test --release`

**Comment by tomusdrw** [comment]
This is the best I came up with. 
1. Clippy dependency is optional (and has to be because it's not possible to compile clippy on stable/beta).
2. When compiling on nightly for development (without `--release` flag) the build will fail because of missing `clippy` create.
3. You have to explicitly invoke `cargo test --features dev-clippy` when building without `--release` on nightly.
4. It's not possible to add dependency or to enable feature from `build.rs`

File: Cargo.toml [hunk 2]
```diff
@@ -12,22 +16,30 @@ rustc-serialize = "0.3"
...
 
 [features]
 default = ["rpc"]
 rpc = ["ethcore-rpc"]
-dev = ["clippy", "ethcore/dev", "ethcore-util/dev", "ethsync/dev", "ethcore-rpc/dev"]
+dev = ["ethcore/dev", "ethcore-util/dev", "ethsync/dev", "ethcore-rpc/dev"]
+dev-clippy = ["clippy", "ethcore/clippy", "ethcore-util/clippy", "ethsync/clippy", "ethcore-rpc/clippy"]
 travis-beta = ["ethcore/json-tests"]
-travis-nightly = ["ethcore/json-tests", "dev"]
+travis-nightly = ["ethcore/json-tests", "clippy", "dev"]
```
Comment (debris) [inline Cargo.toml]: I think, it should be `dev-clippy`.

**Comment by gavofyork** [comment]
does this address the problem properly?

**Comment by tomusdrw** [comment]
It just forces you to use `--feature dev-clippy` when compiling on nightly without `--release`. I'm not sure if that's the correct way to address the problem.

There is also a convenience script included to run something like `./cargo.sh test -p ethcore` (it adds the feature for you).

**Pull request merged**